app/web/src/components/UserProfile/UserProfile.tsx

```diff
@@ -65,44 +73,45 @@ const UserProfile = ({
             {/* Side panel */}
             <section className="bg-ch-gray-760 font-fira-sans p-12 md:overflow-y-auto ch-scrollbar">
               <div className="flex gap-6">
-                {!isEditable && (
+                {!isEditing && (
                   <div className="w-28 flex-shrink-0">
-                    <fields.image.component
+                    <ImageField
                       field={fields.image}
+                      dispatch={fieldDispatch}
                       user={user}
                       save={onSave}
-                      hasEditPermission={hasEditPermission}
+                      hasPermissionToEdit={hasPermissionToEdit}
                     />
```

Comment:
Check this out @Irev-Dev, I think I found something way better here. Using components instead of that big old config object.